### DISCUSSION TITLE
Making an SVG scale

### DISCUSSION THREAD

**Discussion by memsom** [Mapsui/Mapsui#1719] [Q&A]
Is there a way to make a feature, in my example an SVG keep its own scale? I was looking at the examples and I can see all the features based on polygons and lines will fix to a specific size and scale with the map. I have a requirement to allow adding images that have a size and scale with the map. Is that currently possible?

**Comment by pauldendulk** [comment]
No, you would have to convert to the SVG to a NTS Geometry.

**Reply by memsom** [reply]
Is there an easy way to do that?

Can we maybe instead scale the SVG using the zoom level of the map? I guess it is proportional and as the map layers are rerendered the scale can be recalculated?

**Reply by charlenni** [reply]
Perhaps you could make your own renderer. Look like the bitmaps are rendered. There are also SVGs. Perhaps you could add a flag, which decides, if the SVG should be scaled with the map.

**Reply by charlenni** [reply]
SVG renderer take the Scale into account. So you could check, if the resolution of the map changes. If this is the case, change all the SymbolScales of your SVGs. Perhaps this works.

**Reply by memsom** [reply]
> SVG renderer take the Scale into account. So you could check, if the resolution of the map changes. If this is the case, change all the SymbolScales of your SVGs. Perhaps this works.

Yep, I can confirm that this works. The only weird part was mapping the resolution of the ViewPort to a scale. I wanted to use a zoom level (0..20) as it is more human readable, and that sort of works okay, but the quick POC I made is a bit bumpy when scaling the image. It appears that Mapsui actually deals with partial zooms, as in it seems to start up at circa 1.5 (this is my calculated value). I assumed it would only be actual zoom levels as you get from the map's Resolutions array. No matter. 

Looking again, I guess I smoothed out the scaling too much and I think that is why it is jerky. I pinned it to .0/.15/.25/.50/.75. 

All our code wraps Mapsui, as we got previously bitten by using a specific Map Control. If I can clean it up and make it more generic, I will submit a PR.

**Reply by pauldendulk** [reply]
Mapsui's resolution is continuous. The 'levels' are just a number of stops which can be used (for instance by +/- buttons) but will be ignored in other situations. So you need to work with the resolution value.
